src/core/tests/Test_ZypperPackageManager.py

```diff
@@ -183,5 +184,42 @@ def test_install_package_failure(self):
         # test for unsuccessfully installing a package
         self.assertEquals(package_manager.install_update_and_dependencies('selinux-policy.noarch', '3.13.1-102.el7_3.16', simulate=True), Constants.FAILED)
 
+    def test_get_process_tree_from_package_manager_output_success(self):
+        self.runtime.set_legacy_test_type('HappyPath')
+
+        package_manager = self.container.get('package_manager')
+        self.assertIsNotNone(package_manager)
+
+        # Create example package manager message and include test pid
+        package_manager_output = 'Output from package manager: | System management is locked by the application with pid 7914 (/usr/bin/zypper).'
+
+        # Test to make sure a valid string was returned with process information
+        self.assertIsNotNone(package_manager.get_process_tree_from_pid_in_output(package_manager_output))
+
+    def test_get_process_tree_from_package_manager_output_failure_nonexistent_process(self):
+        self.runtime.set_legacy_test_type('SadPath')
```

Comment:
> Please run the 3 tests you have added for this log under debug mode, one at a time and make sure they follow the expected code path within the function

Thanks for this tip, I wasn't familiar with how to debug in pycharm. I've done so and have verified that they are following the expected code path in the function.

Reply:
> Don't see the mock output for get process tree command under SadPath

I didn't add any because it's expected to not have any output. However, after some testing, that actually won't be the case. The process tree command will always return some sort of process tree, even if the pid doesn't belong to a valid process. I've added a search for the pid in the process tree returned to verify that it actually contains the pid used in the command.